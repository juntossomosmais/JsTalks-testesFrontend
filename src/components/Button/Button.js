import React, { useState } from "react";

const Button = () => {
  const [text, setText] = useState("");
  const handleClick = () => {
    setText("Button cliked");
  };
  return <button onClick={handleClick}>{text || props.text}</button>;
};

export default Button;
