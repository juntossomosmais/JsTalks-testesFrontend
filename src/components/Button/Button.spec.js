import React from "react";
import { render, fireEvent } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";

import Button from "./Button";

describe("<Button />", () => {
  it("should return the props text", () => {
    // given
    const expectedResult = "TestButton";

    // when
    const component = render(<Button text={expectedResult} />);

    // then
    expect(component.getByText("TestButton")).toBeInTheDocument();
    expect(component.baseElement).toMatchSnapshot("prop text");
  });
  it("should return the state text", () => {
    // given
    const propText = "TestButton";

    // when
    const { getByText, baseElement, rerender } = render(
      <Button text={propText} />
    );

    fireEvent.click(getByText(propText));
    rerender(<Button text={propText} />);

    // then
    expect(getByText("Button cliked")).toBeInTheDocument();
    expect(baseElement).toMatchSnapshot("state text");
  });
});
