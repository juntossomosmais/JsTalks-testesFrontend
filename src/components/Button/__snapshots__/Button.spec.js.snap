// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> should return the props text: prop text 1`] = `
<body>
  <div>
    <button
      class="sc-AxjAm kwfDhX"
    >
      <span>
        TestButton
      </span>
    </button>
  </div>
</body>
`;

exports[`<Button /> should return the state text: state text 1`] = `
<body>
  <div>
    <button
      class="sc-AxjAm kwfDhX"
    >
      <span>
        Button cliked
      </span>
    </button>
  </div>
</body>
`;
